ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -543,6 +542,16 @@ private void analyzeGroupBy(final GroupBy groupBy) {
       }
     }
 
+    private void analyzePartitionBy(final Expression partitionBy) {
+      if (partitionBy instanceof ColumnReferenceExp) {
```

Comment:
[As discussed](https://github.com/confluentinc/ksql/pull/3982#issuecomment-559452186) this will need to handle expressions, though this could be a follow up PR.

Reply:
Yup, let's do this in a follow-up

Reply:
https://github.com/confluentinc/ksql/issues/4018

Reply:
I've added to ksqlDb v1.0 project as we don't want to release any new version of KSQL till this is done.